backends-velox/src/main/scala/org/apache/gluten/execution/GenerateExecTransformer.scala

```diff
@@ -243,15 +242,15 @@ object PullOutGenerateProjectHelper extends PullOutProjectHelper {
               case jsonPath if jsonPath.foldable =>
                 Option(jsonPath.eval()) match {
                   case Some(path) =>
-                    GetJsonObject(jsonObj, Literal.create(JSON_PATH_PREFIX + path))
+                    GetJsonObject(jsonObj, Literal.create("$[" + path + "]"))
                   case _ =>
                     Literal.create(null)
                 }
               case jsonPath =>
-                // TODO: The prefix is just for adapting to GetJsonObject.
-                // Maybe, we can remove this handling in the future by
-                // making path without "$." recognized
-                GetJsonObject(jsonObj, Concat(Seq(Literal.create(JSON_PATH_PREFIX), jsonPath)))
+                // Build bracket-notation uniformly to support nested paths, e.g., $[a.b]
```

Comment:
I think it doesn't mean nested path, but just the field name containing dot. 

Suggestion:
Build bracket notation uniformly to allow dot-containing field names in JSON paths, e.g., $[a.b]".

Reply:
fixed, thanks~